### DISCUSSION TITLE
Sonoff NS panel won't flash

### DISCUSSION THREAD

**Discussion by bkbartk** [arendst/Tasmota#15034] [Q&A] [answered]
Hello,

I am trying to flash my sonoff ns panel with tasmota,-nspanel, but it fails.
I booted up the device and it was trying to pair with ewelink, so the device works.
I connected my usb to uart to an ESP8266(or something) device and I was able to flash, so that one works too.
But when I try to flash my nspanel it's completely unresponsive.
I connect ground to I00 while booting or the whole time.

I tried the webinstaller. which says:
![image](https://user-images.githubusercontent.com/17070805/156875396-9df5d31b-690e-460b-9238-0daef0443ffd.png)
and esp flasher, (https://github.com/Jason2866/ESP_Flasher/releases)
for windows,
which either gives one of those 2 errors:

> Using 'COM4' as serial port.
Connecting........_____....._____....._____.....____
Unexpected error: Cannot configure port, something went wrong. Original message: PermissionError(13, 'Access is denied.', None, 5)

> Using 'COM4' as serial port.
Connecting........_____....._____....._____....._____....._____....._____....._____
Unexpected error: ESP Chip Auto-Detection failed: Failed to connect to Espressif device: Timed out waiting for packet header

I tried a lot of times, but the same result every time.
can there be an issue with my nspanel?
or do I need to provide 5V during flashing?
as shown here
![image](https://user-images.githubusercontent.com/17070805/156875518-948f961a-7150-4c8e-add4-bb902a2cd598.png)
or am I missing something completely?

**Comment by Jason2866** [comment]
Which USB Serial adapter do you use? For flashing a ESP32 device it has to be one with a onboard 3.3 Vreg.
ESP32 draws more power than a ESP8266. See our recommendation https://tasmota.github.io/docs/Getting-Started/#serial-to-usb-adapter and follow the flash instruction as described here https://templates.blakadder.com/sonoff_NSPanel.html

**Comment by bkbartk** [comment]
it very well could be that my USB to serial adapter is not compatible with ESP32,
this is the fist time I try to flash ESP32.
I use this adapter
https://nl.aliexpress.com/item/32880581501.html?_randl_currency=EUR&_randl_shipto=NL&src=google&src=google&albch=shopping&acnt=494-037-6276&slnk=&plac=&mtctp=&albbt=Google_7_shopping&albagn=888888&isSmbAutoCall=false&needSmbHouyi=false&albcp=6459980570&albag=76980386066&trgt=539263010115&crea=nl32880581501&netw=u&device=c&albpg=539263010115&albpd=nl32880581501&gclid=CjwKCAiAsYyRBhACEiwAkJFKot9yWOLqX7i7X0t3zosID7gCs8YuTfIVX0Af0fr3zgzH8QiXkBkOHhoCE2wQAvD_BwE&gclsrc=aw.ds&aff_fcid=1a73cd35cc044ed6a9b4e7f5a048e5ec-1646505031859-03573-UneMJZVf&aff_fsk=UneMJZVf&aff_platform=aaf&sk=UneMJZVf&aff_trace_key=1a73cd35cc044ed6a9b4e7f5a048e5ec-1646505031859-03573-UneMJZVf&terminal_id=26ed0ae4516e46b68cdff63676e15461&afSmartRedirect=y
for most of my tests
I also tried this one,
https://nl.aliexpress.com/item/1005003599894517.html?_randl_currency=EUR&_randl_shipto=NL&src=google&src=google&albch=shopping&acnt=494-037-6276&slnk=&plac=&mtctp=&albbt=Google_7_shopping&albagn=888888&isSmbAutoCall=false&needSmbHouyi=false&albcp=6459980570&albag=76980386066&trgt=539263010115&crea=nl1005003599894517&netw=u&device=c&albpg=539263010115&albpd=nl1005003599894517&gclid=CjwKCAiAsYyRBhACEiwAkJFKomYHgawoglnJxRAa9qMo9r4JiIPfHmX5BMXGDe-wyDBqxBYCzlsTbxoCSwoQAvD_BwE&gclsrc=aw.ds&aff_fcid=f4ef3346500741688dac3d2b22e7bccc-1646505242852-04651-UneMJZVf&aff_fsk=UneMJZVf&aff_platform=aaf&sk=UneMJZVf&aff_trace_key=f4ef3346500741688dac3d2b22e7bccc-1646505242852-04651-UneMJZVf&terminal_id=26ed0ae4516e46b68cdff63676e15461&afSmartRedirect=y
but I am not completely sure if i can flash other stuff then the ESP32-cam with it,

according to the manual, I think I should just use this adaptor
https://nl.aliexpress.com/item/32761423124.html?gatewayAdapt=glo2nld
or am I mistaken heren?

**Comment by Jason2866** [comment]
Mhh, both have a decent 3.3Vreg on board. Both should work.
Short wires? Good Usb cable? PC USB Port delivers enough current?

**Comment by bkbartk** [comment, marked as answer]
I got it working,
not sure why the "ESP32-CAM-MB" didn't work, but for the "USB to ESP-01 adapter" I used my default pinout.
but it turn our on the Sonoff NSpanel the Rx and TX are swapped compared with the sonoff wall switches I have.

So actually all I needed to do was swapping the wires.

**Reply by gdespirito** [reply]
This happened to me. I swaped TX an RX. Thanks!! You saved me hours

**Reply by mondbaer** [reply]
Thank you. This helped me a lot and saved me a lot of time.

**Reply by t-rekttt** [reply]
OMG you are my hero
